contracts/InterchainTokenService.sol

```diff
@@ -280,52 +283,69 @@ contract InterchainTokenService is
     \************/
 
     /**
-     * @notice Used to deploy remote custom TokenManagers.
-     * @dev At least the `gasValue` amount of native token must be passed to the function call. `gasValue` exists because this function can be
-     * part of a multicall involving multiple functions that could make remote contract calls.
-     * This method is temporarily restricted in the following scenarios:
-     * - Deploying to a remote chain and the destination chain is connected via ITS Hub
-     * - Deploying to the current chain, if connected as an Amplifier chain, i.e existing ITS contracts on consensus chains aren't affected.
-     * Once ITS Hub adds support for deploy token manager msg, the restriction will be lifted.
-     * Note that the factory contract can still call `deployTokenManager` to facilitate canonical token registration.
-     * @param salt The salt to be used during deployment.
-     * @param destinationChain The name of the chain to deploy the TokenManager and standardized token to.
-     * @param tokenManagerType The type of token manager to be deployed. Cannot be NATIVE_INTERCHAIN_TOKEN.
-     * @param params The params that will be used to initialize the TokenManager.
-     * @param gasValue The amount of native tokens to be used to pay for gas for the remote deployment.
-     * @return tokenId The tokenId corresponding to the deployed TokenManager.
+     * @notice Registers metadata for a token on the ITS Hub. This metadata is used for scaling linked tokens.
+     * @param tokenAddress The address of the token.
+     * @param gasValue The gas value for deployment.
      */
-    function deployTokenManager(
+    function registerTokenMetadata(address tokenAddress, uint256 gasValue) external payable {
+        if (tokenAddress == address(0)) revert ZeroAddress();
+
+        uint8 decimals = IERC20Named(tokenAddress).decimals();
+
+        bytes memory payload = abi.encode(MESSAGE_TYPE_REGISTER_TOKEN_METADATA, tokenAddress.toBytes(), decimals);
+
+        string memory destinationAddress = trustedAddress(ITS_HUB_CHAIN_NAME);
+        // Check whether no trusted address was set for ITS Hub chain
+        if (bytes(destinationAddress).length == 0) revert UntrustedChain();
+
+        (bool success, bytes memory returnData) = gatewayCaller.delegatecall(
+            abi.encodeWithSelector(
+                IGatewayCaller.callContract.selector,
+                ITS_HUB_CHAIN_NAME,
+                destinationAddress,
+                payload,
+                IGatewayCaller.MetadataVersion.CONTRACT_CALL,
+                gasValue
+            )
+        );
+
+        if (!success) revert GatewayCallFailed(returnData);
+    }
+
+    function linkToken(
         bytes32 salt,
         string calldata destinationChain,
+        bytes memory destinationTokenAddress,
         TokenManagerType tokenManagerType,
-        bytes calldata params,
+        bytes memory linkParams,
```

Comment:
```suggestion
        bytes calldata linkParams,
```
Only _deployTokenManager needs memory but the remote deploy case can use calldata

Reply:
It increases the compile size a bit, reduced runs